basic_uint.h

```diff
@@ -286,11 +287,51 @@ namespace mgt {
     }
 
     friend basic_uint<l> operator*(const basic_uint<l> &a, const basic_uint<l> &b) {
-      basic_uint<l> res, tmp = b;
-      for (size_t i = 0; i < l; ++i, tmp <<= 1) {
-        if (a.data[i])
-          res += tmp;
+      basic_uint<l> res(0);
+      int current_numa, current_numb;
+      int temp_res;
+      std::vector<int> hexa, hexb;
+      size_t i;
+      for (i = 0; i < l - 5; i += 4) {
+        current_numa = current_numb = 0;
+        for (size_t j = i; j < i + 4; ++j) {
+          if (a.data[j])
+            current_numa += std::pow(2, j - i);
```

Comment:
`std::pow` is slow in this situation, since it can deal with both interger and floating numbers(e.g. `pow(2, 0.5)` returns the square root of 2). i suggest that u implement your own version of pow for integer types.

Reply:
Nice advice.
I'll take it.

Reply:
use fast_pow, which has O(log n) complexity.

Reply:
Just reply directly instead of editing your existing comments. I cannot recieve updates via email if you only edit your comments